src/connection/database/index.js

```diff
@@ -282,8 +282,11 @@ export function databaseConnection(m) {
   return defaultDirectory(m) || defaultDatabaseConnection(m) || l.connection(m, 'database');
 }
 
-export function databaseConnectionName(m) {
-  return (databaseConnection(m) || Map()).get('name');
+export function databaseConnectionName(lock) {
+  const customResolvedConnection = l.resolvedConnection(lock);
+  const standardConnection = databaseConnection(lock);
+  const connection = customResolvedConnection || standardConnection || {};
+  return connection.name;
```

Comment:
You can't directly access `.name` here because it's an Immutable object.
it should be someting like

```js
const connection = customResolvedConnection || standardConnection || new Map();
return connection.get('name');
```

Reply:
.